cmake/modules/BuildDPDK.cmake

```diff
@@ -81,7 +81,7 @@ function(do_build_dpdk dpdk_dir)
 
   include(ExternalProject)
   ExternalProject_Add(dpdk-ext
-    SOURCE_DIR ${CMAKE_SOURCE_DIR}/src/spdk/dpdk
+    SOURCE_DIR ${CMAKE_SOURCE_DIR}/src/seastar/dpdk
```

Comment:
i am not sure if SPDK backend will work or not if we switch over to the seastar/dpdk. if what we are after is just some PMD drivers. we can always re-enable them in `patch-dpdk-conf.sh`. you could add something like
```cmake
set(dpdk_config_args
  ${dpdk_dir} ${machine} ${arch})
if(WITH_SEASTAR)
  list(dpdk_config_args APPEND --with-seastar)
elseif(WITH_SPDK)
  list(dpdk_config_args APPEND --with-spdk)
endif()
ExternalProject_Add_Step(dpdk-ext patch-config
  COMMAND ${CMAKE_MODULE_PATH}/patch-dpdk-conf.sh ${dpdk_config_args}
  DEPENDEES configure
  DEPENDERS build)
```
after updating `patch-dpdk-conf.sh`.

Reply:
Spdk and seastar both have their private modifications to dpdk, see: 
* https://github.com/spdk/dpdk/commits/spdk-19.05
* https://github.com/scylladb/dpdk/commits/local-19.05

So it would be better to switch dpdk `SOURCE_DIR` according to `WITH_SPDK` and `WITH_SEASTAR`.

Currently `patch-dpdk-conf.sh` is working for `seastar/dpdk` without any changes. And maybe I can enable `CONFIG_RTE_EAL_NUMA_AWARE_HUGEPAGES=y` for both spdk and seastar dpdk.